### PR TITLE
For Joins adding function  which can use custom columns to do the join.

### PR DESCRIPTION
**Pull request by fguglani**

### CODE REVIEW

File: src/Sequelize/Models.js [hunk 1]
```diff
@@ -78,3 +78,10 @@ exports._belongsToMany = function (target, source, thru) {
     return null;
   };
 };
+
+exports._belongsToWithFKey = function (target, source, alias, fKey, targetKey) {
```
Comment (voraparth91) [inline src/Sequelize/Models.js:82]: You can take a single json object as "options" and make that have { alias, fKey, targetKey}
This is extendible if you want to include other options in future. Look at the current _belongsTo function, we are making same mistake again!

Comment (fguglani) [reply src/Sequelize/Models.js:82]: Is it okay to change the current instance of _belongsTo as we are not using it as of now in our codeBase.

Comment (voraparth91) [reply src/Sequelize/Models.js:82]: We shouldn't do that. This is our public repo. You never know who else is using this too. We shouldn't break backward compatibility.

Comment (fguglani) [reply src/Sequelize/Models.js:82]: Ya got it

**Pull request merged**